packages/babel-parser/src/parser/error.js

```diff
@@ -73,7 +110,7 @@ export default class ParserError extends CommentsParser {
         }
       }
     }
-    return this._raise({ loc, pos }, message);
+    return this._raise({ code, loc, pos }, message);
```

Comment:
It'd be good to add `params` as a property.
Use case: in Prettier, `Unexpected reserved word 'yield'` should be suppressed, `Unexpected reserved word 'for'` should be rethrown. https://github.com/prettier/prettier/issues/10603

Reply:
I'd prefer to split the error between `UnexpectedKeyword` and `UnexpectedContextualKeyword` if that works too, and avoid exposing too much internal details about how we build errors.

Reply:
@thorn0 I was going to implement this, but I noticed that we already have two different errors:
- `UnexpectedKeyword` for things like `for`, `if`, `while`, ...
- `UnexpectedReservedWord` for things like `yield`, `await`, `enum`, ...